server/src/main/java/org/cloudfoundry/identity/uaa/oauth/UaaTokenStore.java

```diff
@@ -117,10 +118,9 @@ public String createAuthorizationCode(OAuth2Authentication authentication) {
                 }
                 return code;
             } catch (DataIntegrityViolationException exists) {
-                if (tries>=max_tries) throw exists;
+                if (attempt>=maxAttempts) throw exists;
```

Comment:
The throw of `DataIntegrityViolationException` and catch is unnecessary and inefficeint.  Assuming no other lines above in this try block throws `DataIntegrityViolationException`, it can be all done in the same if block, that is:
```
if (updated==0 && attemps>maxAttempt) {
    throw new DataIntegrityViolationException("[oauth_code] Failed to insert code. Result was 0");
}
```

Reply:
Actually, you need to do more than that as you also need to `return code` when `update != 0` but you get the idea.

Reply:
Agreed, the catch and rethrow could be simplified. It wasn't added by the PR, though.

Reply:
I take it back.  The `template.update()` call throws `DataAccessException`, which is a super class of `DataIntegrityViolationException`, so it is possible to get  `DataIntegrityViolationException` thrown from that call, too.  With that said, I am approving this PR as it is.

Reply:
> It wasn't added by the PR, though.

Right, I do know that.